### PR TITLE
feat(indexer): prisma and cli

### PR DESCRIPTION
**Pull request by Larkooo**

### CODE REVIEW

File: crates/indexer/prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    states EntityState[]
+    stateUpdates EntityStateUpdate[]
+    transaction_hash String
+}
+
+model System {
+    id String @id
+    name String
+    calls SystemCall[]
+    queryComponents Component[]
```
Comment (tarrencev) [inline crates/indexer/prisma/schema.prisma:33]:
```suggestion
    query_components Component[]
```

File: crates/indexer/prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    transaction_hash String
+}
+
+model Resource {
+    id String @id
+}
+
+model EntityState {
+    entityId String
```
Comment (tarrencev) [inline crates/indexer/prisma/schema.prisma:42]:
```suggestion
    entity_id String
```

File: crates/indexer/prisma/schema.prisma [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    entity Entity @relation(fields: [entityId], references: [id])
+    componentId String
```
Comment (tarrencev) [inline crates/indexer/prisma/schema.prisma:44]:
```suggestion
    component_id String
```

File: crates/indexer/prisma/schema.prisma [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    transaction_hash String
+}
+
+model Resource {
+    id String @id
+}
+
+model EntityState {
+    entityId String
+    entity Entity @relation(fields: [entityId], references: [id])
```
Comment (tarrencev) [inline crates/indexer/prisma/schema.prisma:43]:
```suggestion
    entity Entity @relation(fields: [entity_id], references: [id])
```

File: crates/indexer/prisma/schema.prisma (hunk 3 above)
Comment (tarrencev) [reply crates/indexer/prisma/schema.prisma:44]: few more below

File: crates/indexer/src/cli.rs [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    world: String,
+    /// The RPC endpoint to use
+    rpc: String,
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let args = Args::parse();
+
+    let world = BigUint::from_str_radix(&args.world[2..], 16).unwrap();
```
Comment (tarrencev) [inline crates/indexer/src/cli.rs:31]: Lets provide some feedback on errors for user provided inputs

For example
```suggestion
    let world = BigUint::from_str_radix(&args.world[2..], 16).unwrap_or_else(|error| {
        panic!("Failed parsing world address: {error:?}");
    });
```

File: crates/indexer/src/cli.rs [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    let client = prisma::PrismaClient::_builder().build().await;
+    assert_eq!(client.is_ok(), true);
+
+    let stream = stream::ApibaraClient::new(rpc).await;
+    match stream {
+        std::result::Result::Ok(s) => {
+            println!("Connected");
+            start(s, client.unwrap(), world).await;
+        }
+        std::result::Result::Err(e) => println!("Error: {:?}", e),
```
Comment (tarrencev) [inline crates/indexer/src/cli.rs:43]: Should we panic here?

File: crates/indexer/src/stream.rs [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+pub enum Chain {
+    AlphaMainnet,
+    AlphaGoerli,
+    AlphaGoerli2,
+}
+
+impl From<Chain> for &'static str {
+    fn from(chain: Chain) -> Self {
+        match chain {
+            Chain::AlphaMainnet => "https://mainnet.starknet.a5a.ch:443",
```
Comment (tarrencev) [inline crates/indexer/src/stream.rs:24]: where are these urls from?

File: crates/dojo-indexer/src/stream.rs [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    #[allow(clippy::needless_lifetimes)]
+    pub async fn request_data<'x>(
+        &'x mut self,
+        filter: Filter,
+    ) -> Result<impl Stream<Item = Result<Option<Data>, tonic::Status>> + 'x, tonic::Status> {
+        self.stream_id += 1;
+        let request = StreamDataRequest {
+            stream_id: Some(self.stream_id),
+            batch_size: Some(1),
+            starting_cursor: Some(Cursor { order_key: 1, unique_key: vec![] }),
```
Comment (fracek) [inline crates/dojo-indexer/src/stream.rs:56]: If you set `starting_cursor` to `None` it will stream from genesis.

Comment (tarrencev) [reply crates/dojo-indexer/src/stream.rs:56]: hm we prob want the start block as part of the world config

**Pull request merged**